tools/generate_linux_syscalls.zig

```diff
@@ -48,710 +48,648 @@ fn isReservedNameOld(name: []const u8) bool {
+    filters: Filters,
+    writer: anytype,
+    optional_writer: anytype,
+) !void {
+    _ = optional_writer;
+
+    var lines = mem.tokenizeScalar(u8, bytes, '\n');
+    while (lines.next()) |line| {
+        if (line[0] == '#') continue;
+
+        var fields = mem.tokenizeAny(u8, line, " \t");
+        const number = fields.next() orelse return error.Incomplete;
+
+        const abi = fields.next() orelse return error.Incomplete;
+        if (filters.abiCheck) |abiCheckFn| {
+            switch (abiCheckFn(abi)) {
+                .none => {},
+                .@"break" => break,
+                .@"continue" => continue,
             }
         }
-
-        try writer.writeAll(
-            \\};
-            \\
-            \\pub const PowerPC64 = enum(usize) {
-            \\
-        );
-        try writer.writeAll(list_64.items);
-        try writer.writeAll("};\n\n");
-    }
-    {
-        try writer.writeAll("pub const S390x = enum(usize) {\n");
-
-        const table = try linux_dir.readFile("arch/s390/kernel/syscalls/syscall.tbl", buf);
-        var lines = mem.tokenizeScalar(u8, table, '\n');
-        while (lines.next()) |line| {
-            if (line[0] == '#') continue;
-
-            var fields = mem.tokenizeAny(u8, line, " \t");
-            const number = fields.next() orelse return error.Incomplete;
-            const abi = fields.next() orelse return error.Incomplete;
-            if (mem.eql(u8, abi, "32")) continue; // 32-bit s390 support in linux is deprecated
-            const name = fields.next() orelse return error.Incomplete;
-            const fixed_name = if (stdlib_renames.get(name)) |fixed| fixed else name;
-
-            try writer.print("    {p} = {s},\n", .{ zig.fmtId(fixed_name), number });
+        const name = fields.next() orelse return error.Incomplete;
+        if (filters.isReservedNameOld) |isReservedNameOldFn| {
+            if (isReservedNameOldFn(name)) continue;
         }
+        const fixed_name = if (filters.fixedName) |fixedNameFn| fixedNameFn(name) else name;
 
-        try writer.writeAll("};\n\n");
+        try writer.print("    {p} = linux_base + {s},\n", .{ zig.fmtId(fixed_name), number });
     }
-    {
-        try writer.writeAll("pub const Xtensa = enum(usize) {\n");
-
-        const table = try linux_dir.readFile("arch/xtensa/kernel/syscalls/syscall.tbl", buf);
-        var lines = mem.tokenizeScalar(u8, table, '\n');
-        while (lines.next()) |line| {
-            if (line[0] == '#') continue;
-
-            var fields = mem.tokenizeAny(u8, line, " \t");
-            const number = fields.next() orelse return error.Incomplete;
-            // abi is always common
-            _ = fields.next() orelse return error.Incomplete;
-            const name = fields.next() orelse return error.Incomplete;
-            if (isReservedNameOld(name)) continue;
-            const fixed_name = if (stdlib_renames.get(name)) |fixed| fixed else name;
+}
 
+fn processPowerPcBasedArch(
```

Comment:
This was a cute trick when I wrote it, but looking back I think it'd be better to split the logic for ppc32/6 and read the same file twice. It's not like this is performance critical.

Reply:
Any reasons on why it s a bad idea ? when i've worked on it other architectures can benefit from it like `sparc/sparc64` which gets it's values from the same file `arch/sparc/kernel/syscalls/syscall.tbl`, but yeah it is not performance critical if you want me to separate them I can do that